cryptoki/src/types/slot_token.rs

```diff
@@ -42,3 +42,45 @@ impl From<Slot> for CK_SLOT_ID {
         slot.slot_id
     }
 }
+
+///
+#[derive(Debug, Clone, Copy, Default)]
+#[repr(C)]
+pub struct TokenInfo {
+    ///
+    pub label: [u8; 32],
+    ///
+    pub manufacturer_id: [u8; 32],
+    ///
+    pub model: [u8; 16],
+    ///
+    pub serial_number: [u8; 16],
+    ///
+    pub flags: u32,
+    ///
+    pub max_session_count: u32,
+    ///
+    pub session_count: u32,
+    ///
+    pub max_rw_session_count: u32,
+    ///
+    pub rw_session_count: u32,
+    ///
+    pub max_pin_len: u32,
+    ///
+    pub min_pin_len: u32,
+    ///
+    pub total_public_memory: u32,
+    ///
+    pub free_public_memory: u32,
+    ///
+    pub total_private_memory: u32,
+    ///
+    pub free_private_memory: u32,
```

Comment:
I think these should be using [`Ulong`](https://docs.rs/cryptoki/0.1.1/cryptoki/types/struct.Ulong.html)?

Reply:
Replaced all that with inner `CK_TOKEN_INFO`. Thanks for the idea!